src/notation/internal/notationinteraction.cpp

```diff
@@ -689,24 +689,72 @@ void NotationInteraction::moveChordNoteSelection(MoveDirection d)
         return;
     }
 
-    EngravingItem* current = selection()->element();
-    if (!current || !(current->isNote() || current->isRest())) {
-        return;
-    }
+    auto removeDuplicates = [](std::vector<EngravingItem*>&elements) {
+        std::sort(elements.begin(), elements.end());
+        auto it = std::unique(elements.begin(), elements.end());
+        elements.erase(it, elements.end());
+    };
 
-    EngravingItem* chordElem;
-    if (d == MoveDirection::Up) {
-        chordElem = score()->upAlt(current);
-    } else {
-        chordElem = score()->downAlt(current);
+    auto& _score = *score();
+    auto& selection = _score.selection();
+    EngravingItem* currentSingle = selection.element();
+    EngravingItem* oldSingle = currentSingle;
+    bool isRange = selection.isRange();
+    std::vector<EngravingItem*> notes;
+
+    // Single traverse:
+    if (currentSingle && (currentSingle->isNote() || currentSingle->isRest())) {
+        DirectionV dir = d == MoveDirection::Up ? DirectionV::UP : DirectionV::DOWN;
+        EngravingItem* newSingle = _score.moveAlt(currentSingle, dir);
+        if (newSingle == currentSingle)
+            return;
+        while (newSingle && newSingle->isRest() && toRest(newSingle)->isGap()) {
+            newSingle = _score.moveAlt(newSingle, dir);
+            if (newSingle == oldSingle)
+                break;
+            }
+        select({ newSingle }, SelectType::SINGLE, newSingle->staffIdx());
+        showItem(newSingle);
+    }
+    // Range/List traverse:
+    else for (auto item : selection.elements()) {
+        if (item->isNote()) {
+            auto selectedNote = toNote(item);
+            if (isRange) {
+                auto newSelection = (d == MoveDirection::Down)
+                    ? selectedNote->chord()->downNote()
+                    : selectedNote->chord()->upNote();
+                notes.emplace_back(newSelection);
+            }
+            else { // List
+                DirectionV dir = d == MoveDirection::Up ? DirectionV::UP : DirectionV::DOWN;
```

Comment:
One more squash maybe? you could change moveAlt to take a MoveDirection so that it was a little cleaner

Reply:
But whatever, it doesn't matter. i wonder if it works....

Reply:
Seems not an easy change

Reply:
Let's fix the codestyle stuff first
And check whether it works as intended ;-)